python_modules/libraries/dagster-shared/dagster_shared/error.py

```diff
@@ -30,18 +31,14 @@ class DagsterUnresolvableSymbolError(DagsterError):
 # but that is a significant refactor. Currently we're just moving SerializableErrorInfo so that we
 # can use it with the dg packages`.
 @whitelist_for_serdes
-class SerializableErrorInfo(
-    NamedTuple(
-        "SerializableErrorInfo",
-        [
-            ("message", str),
-            ("stack", Sequence[str]),
-            ("cls_name", Optional[str]),
-            ("cause", Any),
-            ("context", Any),
-        ],
-    )
-):
+@record_custom
```

Comment:
I think you can get by with plain `@record`

* set default val on field decl ie `context: Optional["SerializableErrorInfo"] = None`
* `@record(kw_only=False)` to support var args

Reply:
I still subclass IHAveNew though?

Reply:
i guess not, given the name